arcaflow_plugin_stressng/stressng_schema.py

```diff
@@ -203,318 +149,296 @@ class StressNGParams:
-            "id": "user-time",
-            "name": "CPU User Time",
-            "description": "The CPU time spent in user space",
-        }
-    )
-    system_time: float = dataclasses.field(
-        metadata={
-            "id": "system-time",
-            "name": "CPU System Time",
-            "description": "The CPU time spent in kernel space",
-        }
-    )
-    cpu_usage_per_instance: float = dataclasses.field(
-        metadata={
-            "id": "cpu-usage-per-instance",
-            "name": "CPU usage per instance",
-            "description": (
-                "is the amount of CPU " "used by each stressor instance"
-            ),
-        }
-    )
+    stressor: typing.Annotated[
+        str,
+        schema.name("Stressor"),
+        schema.description("Type of stressor for workload"),
+    ]
+
+    max_rss: typing.Annotated[
+        int,
+        schema.id("max-rss"),
+        schema.name("Max RSS"),
+        schema.description("Maximum resident set size"),
+    ]
+
+    bogo_ops: typing.Annotated[
+        int,
+        schema.id("bogo-ops"),
+        schema.name("Bogus Operations"),
+        schema.description("Number of stressor loop iterations"),
+    ]
+
+    bogo_ops_per_second_usr_sys_time: typing.Annotated[
+        float,
+        schema.id("bogo-ops-per-second-usr-sys-time"),
+        schema.name("Bogus operations per second per user and sys time"),
+        schema.description(
+            "is the bogo-ops rate divided by the user + system time."
+            "This is the real per CPU throughput "
+            "taking into consideration "
+            "all the CPUs used and all the time consumed "
+            "by the stressor and kernel time."
+        ),
+    ]
+
+    bogo_ops_per_second_real_time: typing.Annotated[
+        float,
+        schema.id("bogo-ops-per-second-real-time"),
+        schema.name("Bogus operations per second in real time"),
+        schema.description(
+            "real time measurement is how long the run took based "
+            "on the wall clock time "
+            "(that is, the time the stressor took to run)."
+        ),
+    ]
+
+    wall_clock_time: typing.Annotated[
+        float,
+        schema.id("wall-clock-time"),
+        schema.name("Wall Clock Time"),
+        schema.description("The time the stressor took to run"),
+    ]
+
+    user_time: typing.Annotated[
+        float,
+        schema.id("user-time"),
+        schema.name("CPU User Time"),
+        schema.description("The CPU time spent in user space"),
+    ]
+
+    system_time: typing.Annotated[
+        float,
+        schema.id("system-time"),
+        schema.name("CPU System Time"),
+        schema.description("The CPU time spent in kernel space"),
+    ]
+
+    cpu_usage_per_instance: typing.Annotated[
+        float,
+        schema.id("cpu-usage-per-instance"),
+        schema.name("CPU usage per instance"),
+        schema.description("is the amount of CPU used by each stressor instance"),
+    ]
```

Comment:
For starters, the description should start with a capital letter and not the word "is".

Beyond those, this number appears to be a percentage which is obtained by taking the total CPU time (user + system), dividing it by the number of CPUs, and dividing that by the duration of the test.

So, a better way to express this would be, "System CPU utilization" or "Average per-CPU utilization" (depending on whether you think a fully-engaged system with four CPUs should be 100% or 400% utilized, respectively).

Reply:
Good catch. I think this plugin actually has a number of similar issues like this in the schema. My focus of this PR was to simplify the structure and update the formatting, but I haven't gone through it with a fine-toothed comb to catch and correct all of these text formatting and language issues.